### ISSUE TITLE
No report on pull requests from forks

### ISSUE THREAD

**Issue by axel-op** [axel-op/dart_package_analyzer#2]
## Description
When the action is triggered by the `pull_request` event, if the pull request is from a different repository (a fork), the action fails with the error "Resource not accessible by integration".

EDIT: this behavior has changed. See the comment below.

## Why is this happening
This action uses the [`GITHUB_TOKEN`](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token) you provide to call the GitHub API and to post the result of its analysis.
As explained [here](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#permissions-for-the-github_token), [here](https://github.community/t5/GitHub-Actions/Github-Workflow-not-running-from-pull-request-from-forked/m-p/33547/highlight/true#M1555), or [here](https://github.com/actions/labeler/issues/12), GitHub for now reduces the permissions of this token when the action is triggered by another repository. 
Therefore an error is thrown when this action tries to call the API in this situation.

![Permissions for the GITHUB_TOKEN](https://user-images.githubusercontent.com/49279289/72079867-fcf02c80-32fb-11ea-9625-4ca03991b072.png)

## Current workaround
A possible workaround is to edit your workflow to trigger this action only when pull requests come from the same repository. This can be done by adding an if-condition, as illustrated below:

```yml
name: Example workflow
on: [push, pull_request]

jobs:

  package-analysis:
    if: github.event_name != 'pull_request' || github.actor == '[username]' # Replace [username] by the username of your repo
    runs-on: ubuntu-latest
    steps:
      - ...
```

## 
I will update this issue if there is new information on this.

**Comment by axel-op** [comment]
I edited the code so that this action won't fail anymore on this error.
The report will now be skipped if it cannot be posted.
~~This action will fail if a compilation error or a static error is detected by the analyzer.~~
The outputs of this action will still be set.

**Comment by vaind** [comment]
> The report will now be skipped if it cannot be posted.

Any chance the report could be printed directly to the standard ouptut in that case? Otherwise, if there's a failure later in the pipeline (e.g. our CI fails if the score decreases) we can't see what's wrong.

**Comment by axel-op** [comment]
Hello @vaind, you can still see the original raw JSON report in the logs, along with the values of the outputs. Is this what you want?

![image](https://user-images.githubusercontent.com/49279289/108496556-823d6300-72aa-11eb-8a19-2614d6f3b7b9.png)
![image](https://user-images.githubusercontent.com/49279289/108496628-a00ac800-72aa-11eb-9950-d261250df219.png)

**Comment by vaind** [comment]
Totally missed that, all good then

**Comment by GroovinChip** [comment]
For cases like this, is there any way the raw json from the job logs can be captured and turned into a report or downloadable job artifact? Having to hunt through the logs to see the analysis in these cases is a bit tedious.

**Comment by axel-op** [comment]
> For cases like this, is there any way the raw json from the job logs can be captured and turned into a report or downloadable job artifact? Having to hunt through the logs to see the analysis in these cases is a bit tedious.

Hi @GroovinChip, could you please create a separate issue for this ? :)

**Comment by GroovinChip** [comment]
> > For cases like this, is there any way the raw json from the job logs can be captured and turned into a report or downloadable job artifact? Having to hunt through the logs to see the analysis in these cases is a bit tedious.
> 
> Hi @GroovinChip, could you please create a separate issue for this ? :)

Done
